src/main/scala/io/qbeast/catalog/QbeastCatalogUtils.scala

```diff
@@ -110,21 +106,15 @@ object QbeastCatalogUtils extends Logging {
     }
   }
 
-  private def verifySchema(
-      spark: SparkSession,
-      fs: FileSystem,
-      path: Path,
-      table: CatalogTable): CatalogTable = {
+  private def verifySchema(indexedTable: IndexedTable, table: CatalogTable): CatalogTable = {
+
+    val indexedTableExists = indexedTable.exists
 
-    val isTablePopulated = table.tableType == CatalogTableType.EXTERNAL && fs
-      .exists(path) && fs.listStatus(path).nonEmpty
-    // Users did not specify the schema. We expect the schema exists in Delta.
     if (table.schema.isEmpty) {
+      // Users did not specify the schema. We expect the schema exists
       if (table.tableType == CatalogTableType.EXTERNAL) {
-        if (fs.exists(path) && fs.listStatus(path).nonEmpty) {
-          val existingSchema =
-            DeltaLog.forTable(spark, path.toString).unsafeVolatileSnapshot.metadata.schema
```

Comment:
oh that was a good catch.. why didn't we detected the DeltaLog import before?

Reply:
Good question, the file ended in the src/ folder were we have Delta, so it didn't raise any exception